doc/src/quickstart.md

```diff
@@ -45,6 +55,55 @@ fn main() -> Result<(), Box<std::error::Error>> {
 }
 ```
 
+In this example we are using `CustomAction` to write a simple calculator
+grammar that calculates user input as it is being parsed. We define lexing and
+grammar rules as follows, and store them in `calc.l` and `calc.y` respectively:
+
+```
+%%
+[0-9]+ "INT"
+\+ "PLUS"
+\* "MUL"
+\( "LBRACK"
+\) "RBRACK"
+[\t ]+ ;
+```
+
+```
+%start Expr
+%type MYTYPE
+%%
+Expr: Term 'PLUS' Expr { add($1, $3) }
+    | Term { $1 }
+    ;
+
+Term: Factor 'MUL' Term { mul ($1, $3) }
+    | Factor { $1 }
+    ;
+
+Factor: 'LBRACK' Expr 'RBRACK' { $2 }
+      | 'INT' { int($1) }
+      ;
+%%
+
+type MYTYPE = u64;
+
+fn int(s: &str) -> MYTYPE {
```

Comment:
Add a brief comment here saying something like "this function is in scope for all the grammar actions above"?